src/Products/PluginIndexes/DateRangeIndex/DateRangeIndex.py

```diff
@@ -238,12 +228,24 @@ def uniqueValues(self, name=None, withLengths=0):
                     else:
                         yield (key, len(value))
 
-    def _cache_key(self, catalog):
-        cid = catalog.getId()
-        counter = getattr(aq_base(catalog), 'getCounter', None)
-        if counter is not None:
-            return '%s_%s' % (cid, counter())
-        return cid
+    def _record_cache_key(self, record, resultset=None):
+        term = self._convertDateTime(record.keys[0])
+
+        # why flatten to 10 minutes? 'term' is already flattened
+        # to minutes by _convertDateTime.
+        # tid = isinstance(term, int) and term / 10 or 'None'
```

Comment:
IMO, if a request takes longer than a couple of seconds,  it should be normally an exception. The potential loss of a cached data and time performance is minimal (<100ms) in comparison with the complete duration of a long running request (>1s). The discretization of the index values underlies the same problematic for a long running request, isn't it? In addition, we have a problem with the tests, i.e. [DateRangeIndex/tests.py#L105](https://github.com/zopefoundation/Products.ZCatalog/blob/master/src/Products/PluginIndexes/DateRangeIndex/tests.py#L105). The passed times to the function _apply_index are only between the range [-1,15]. The round mechanism (10 min) collect most of them in the same class and the tests fail.